### DISCUSSION TITLE
Why are there so many indexes for the MIRACL-Korean index?

### DISCUSSION THREAD

**Discussion by steven-channel** [castorini/pyserini#1903] [Q&A]
I'm running the command for evaluating the mDPR model on the MIRACL-Korean benchmark provided here: https://castorini.github.io/pyserini/2cr/miracl.html

The command is:

```
python -m pyserini.search.faiss \
  --threads 16 --batch-size 512 \
  --encoder-class auto \
  --encoder castorini/mdpr-tied-pft-msmarco-ft-miracl-ko \
  --topics miracl-v1.0-ko-dev \
  --index miracl-v1.0-ko-mdpr-tied-pft-msmarco-ft-miracl-ko \
  --output run.miracl.mdpr-tied-pft-msmarco-ft-miracl.ko.dev.txt --hits 1000
```

As far as I know, the number of documents (i.e., positive and negative passages) in the MIRACL-Korean dev set is 3,057 with some duplicates. However, I'm noticing that the number of documents contained in the index is 1,486,752. Why are there so many?

**Comment by GioiaZheng** [comment]
The 1,486,752 count is expected. The dense index is built over the full MIRACL Korean corpus, not just over the documents that appear in the Korean dev qrels.

In that command:

- `--topics miracl-v1.0-ko-dev` selects the dev queries;
- `--index miracl-v1.0-ko-mdpr-tied-pft-msmarco-ft-miracl-ko` selects the prebuilt Faiss index for the full Korean retrieval corpus;
- evaluation then compares the retrieved ranking against the dev relevance judgments.

The qrels only tell `trec_eval` which corpus documents are judged relevant for the dev queries. They are not the candidate set searched by Faiss. Unjudged corpus documents still remain in the index and can be retrieved; they simply do not receive positive relevance credit during evaluation.

This is also why the same Korean corpus has several prebuilt indexes: they use the same `miracl-v1.0-ko` corpus, but different encoders or fine-tuning recipes. The metadata for the index in your command points back to `texts: miracl-v1.0-ko` and lists `documents: 1486752`, which matches what you observed.
